Creating LB object with missing network object crashes metalnet

**Describe the bug**
When creating LB k8s object from yaml file with missing reference Network k8s object, metalnet crashes with error:
```
2023-10-27T09:04:33+02:00       INFO    Observed a panic in reconciler: runtime error: invalid memory address or nil pointer dereference   {"controller": "loadbalancer", "controllerGroup": "networking.metalnet.onmetal.de", "controllerKind": "LoadBalancer", "LoadBalancer": {"name":"lb2","namespace":"default"}, "namespace": "default", "name": "lb2", "reconcileID": "5e9ed3c3-f9f0-4edb-9550-2386777baeb3"}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x10bbf80]

goroutine 90 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
        /home/vilo/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:119 +0x1a4
panic({0x1259720, 0x2338f90})
        /usr/local/go/src/runtime/panic.go:884 +0x1f4
github.com/onmetal/metalnet/controllers.(*LoadBalancerReconciler).reconcile(0x4000470af0, {0x16fb088, 0x40002c0e10}, {{0x16fd488?, 0x40002c0e40?}, 0x4000042a56?}, 0x40001a8a80)
        /home/vilo/go/metalnet/controllers/loadbalancer_controller.go:246 +0x2d0
github.com/onmetal/metalnet/controllers.(*LoadBalancerReconciler).reconcileExists(0x16fb088?, {0x16fb088, 0x40002c0e10}, {{0x16fd488?, 0x40002c0e40?}, 0x4000042a56?}, 0x40001a8a80)
        /home/vilo/go/metalnet/controllers/loadbalancer_controller.go:90 +0x6c
github.com/onmetal/metalnet/controllers.(*LoadBalancerReconciler).Reconcile(0x4000470af0, {0x16fb088, 0x40002c0e10}, {{{0x4000042a59?, 0x30?}, {0x4000042a56?, 0xffff797c5e01?}}})
        /home/vilo/go/metalnet/controllers/loadbalancer_controller.go:78 +0x1ec
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x16fb088?, {0x16fb088?, 0x40002c0e10?}, {{{0x4000042a59?, 0x11cf9e0?}, {0x4000042a56?, 0x0?}}})
        /home/vilo/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:122 +0x8c
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0x400019d7c0, {0x16fafe0, 0x4000284190}, {0x12e2300?, 0x4000449b40?})
        /home/vilo/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:323 +0x2cc
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0x400019d7c0, {0x16fafe0, 0x4000284190})
        /home/vilo/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:274 +0x1a0
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
        /home/vilo/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:235 +0x74
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
        /home/vilo/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:231 +0x43c 
```
Existing automated test doesn't have this behaviour (LB object is created, reconciled and in Pending state).

**To Reproduce**
run metalnet and create LB object from yaml file with network in NetworkRef that doesn't exist.

**Additional context**
When I commented out line pointed in error trace:
`r.Eventf(lb, corev1.EventTypeWarning, "NetworkNotFound", "Network %s could not be found", networkKey.Name)`
LB object was created and reconciled as in test.

--- a/main.go
+++ b/main.go
@@ -291,13 +291,14 @@ func main() {
 	}
 
 	if err = (&controllers.LoadBalancerReconciler{
-		Client:     mgr.GetClient(),
-		Scheme:     mgr.GetScheme(),
-		DPDK:       dpdkclient.NewClient(dpdkProtoClient),
-		Metalbond:  metalbond.NewClient(mbInstance),
-		NodeName:   nodeName,
-		PublicVNI:  publicVNI,
-		MBInternal: mbClient,
+		Client:        mgr.GetClient(),
+		Scheme:        mgr.GetScheme(),
+		EventRecorder: mgr.GetEventRecorderFor("loadbalancer"),
+		DPDK:          dpdkclient.NewClient(dpdkProtoClient),
+		Metalbond:     metalbond.NewClient(mbInstance),
+		NodeName:      nodeName,
+		PublicVNI:     publicVNI,
+		MBInternal:    mbClient,
 	}).SetupWithManager(mgr); err != nil {
 		setupLog.Error(err, "unable to create controller", "controller", "LoadBalancer")
 		os.Exit(1)

